### PR TITLE
Skavya/sram stub

### PR DESCRIPTION
**Pull request by kavyasreedhar**

### CODE REVIEW

File: lake/top/lake_top.py [hunk 1]
```diff
@@ -51,7 +53,7 @@ def __init__(self,
                  config_data_width=16,
                  config_addr_width=8,
                  remove_tb=False,
-                 fifo_mode=True):
+                 fifo_mode=False):
```
Comment (mbstrange2) [inline lake/top/lake_top.py]: this should stay as true

Comment (kavyasreedhar) [reply lake/top/lake_top.py]: Changed

File: lake/passes/passes.py [hunk 2]
```diff
@@ -47,3 +48,49 @@ def visit(self, node):
...
+
+        class SRAMPortNames(IRVisitor):
+            def __init__(self, use_sram_stub, ports):
+                IRVisitor.__init__(self)
+                self.use_sram_stub = use_sram_stub
+                self.ports = ports
+
+            def visit(self, node):
+                if isinstance(node, Port):
+                    if not(len(node.get_attributes()) <= 0 or self.use_sram_stub):
```
Comment (Kuree) [inline lake/passes/passes.py]: try `if node.has_attribute("sram_port")` and use that to replace line 74-76

File: lake/passes/passes.py [hunk 3]
```diff
@@ -47,3 +48,49 @@ def visit(self, node):
...
+                        for i in range(len(node.get_attributes())):
+                            if (isinstance(node.get_attributes()[i].get(), SRAMPortAttr)):
+                                if node.name == "sram_addr":
+                                    node.name = self.ports[0]
```
Comment (Kuree) [inline lake/passes/passes.py]: Why can't you use `node.name = self.addr_port_name`?

Comment (kavyasreedhar) [reply lake/passes/passes.py]: Oh it does the same thing, but your way is more readable, so I will change it.

File: lake/passes/passes.py (hunk 2 above)
Comment (kavyasreedhar) [reply lake/passes/passes.py]: Changed, thanks.

File: utils/sram_macro.py [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+    def set_name(self, name):
+        self.name = name
+
+    def set_addr_port_name(self, addr_port):
+        self.addr_port_name = addr_port
+        self.ports[0] = addr_port
+
+    def set_ce_port_name(self, ce_port):
+        self.ce_port_name = ce_port
+        self.ports[1] = ce_port
```
Comment (Kuree) [inline utils/sram_macro.py]: These can be removed?

Comment (kavyasreedhar) [reply utils/sram_macro.py]: yup, deleted.

File: utils/flattenND.py [hunk 5]
```diff
@@ -12,8 +12,8 @@ def __init__(self, array_width, *array_size):
             total_dims = total_dims * array_size[i]
             apparent_width = apparent_width * array_size[i]
 
-        self.in_ = self.input("in", width=array_width, size=array_size)
-        self.out_ = self.output("out", width=total_dims)
+        self.in_ = self.input("input_array", width=array_width, size=array_size, packed=True)
```
Comment (mbstrange2) [inline utils/flattenND.py:15]: if you use packed array, please use explicit_array as well for consistency with rest of source

File: utils/reverse_flatten.py [hunk 6]
```diff
@@ -12,8 +12,8 @@ def __init__(self, array_width, *array_size):
             total_dims = total_dims * array_size[i]
             apparent_width = apparent_width * array_size[i]
 
-        self.in_ = self.input("in", width=total_dims)
-        self.out_ = self.output("out", width=array_width, size=array_size)
+        self.in_ = self.input("input_array", width=total_dims)
+        self.out_ = self.output("output_array", width=array_width, size=array_size, packed=True)
```
Comment (mbstrange2) [inline utils/reverse_flatten.py:16]:
```suggestion
        self.out_ = self.output("output_array", width=array_width, size=array_size, packed=True, explicit_array=True)
```

File: utils/flattenND.py (hunk 5 above)
Comment (kavyasreedhar) [reply utils/flattenND.py]: Done

File: utils/reverse_flatten.py (hunk 6 above)
Comment (kavyasreedhar) [reply utils/reverse_flatten.py]: Done

**Pull request merged**